catalog/core/catalog-core-api/src/main/java/ddf/catalog/source/CatalogStore.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.source;
+
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.operation.UpdateResponse;
+
+/**
+ * A CatalogStore represents an editable store that is either local or remote
+ *
+ */
+public interface CatalogStore extends FederatedSource {
```

Comment:
We wanted it to be distinct from CatalogProvider. We didn't want CatalogStores to be grouped in with the CatalogProvider since we deal with them differently. CatalogProvider is treated a local and CatalogStore is treated as remote.